doc/balancer_simulation.md

```diff
@@ -0,0 +1,124 @@
+# Balancer Protocol Analysis Report
+
+## Purpose
+
+The purpose of this report is to evaluate the viability of withdrawing all assets from a Balancer pool with additional incentives to facilitate the pool's liquidity. This is crucial for a liquidity provider looking to fully divest from a pool. Our findings reveal deviations from expectations based on assumptions outlined in Balancer's white paper.
+
+## Weighted Pool Simulation
+
+### Scenario 1
+
+Let's create a scenario where a pool is comprised of two tokens: X and Y. The initial quantities of these tokens are `X = 1000, Y = 1000`. Thus, the Initial Liquidity Provider (LP) token supply is `2000 (1000X + 1000Y)`.
+
+Suppose another user wants to deposit the same amount of tokens into this pool (`Xd = 1000, Yd = 1000`).
+
+According to Balancer's multi-deposit formula (`Dk = ((Ps - Pi)/Ps - 1) * Bk`), we need to determine the quantity of LP tokens to mint for this deposit. The variables in the formula are defined as follows:
+
+- Dk: The deposit amount of token k
+- Bk: The amount of token k in the pool
+- Pi: Newly issued LP token amount
+- Ps: Current total supply of LP tokens
+
+From this formula, we derive the quantity of newly issued LP tokens as `Pi = Ps * Dk/Bk`.
+
+Thus, the minted LP tokens are:
+
+- For X tokens: `Pix = 2000 * 1000/1000 = 2000`
+- For Y tokens: `Piy = 2000 * 1000/1000 = 2000`
+
+As a result, the total supply of LP tokens is `Ps = 2000 + 2000 + 2000 = 6000` with `Bx = 2000, By = 2000`.
```

Comment:
With the new formula, we will get correct LP tokens  `Ps = 2000 + 1000 + 1000 = 4000`

Once we have correct amount of LP tokens, the losses will disappear.